axelor-stock/src/main/java/com/axelor/apps/stock/service/StockMoveLineServiceImpl.java

```diff
@@ -536,8 +543,14 @@ public StockMoveLine getMergedStockMoveLine(List<StockMoveLine> stockMoveLineLis
 			quantity = quantity.add(stockMoveLine.getQty());
 		}
 
-		return createStockMoveLine(product, productName, description, quantity, unitPriceUntaxed, unitPriceTaxed, unit,
-				stockMove, trackingNumber);
+		StockMoveLine stockMoveLine = null;
+		try {
+			stockMoveLine = createStockMoveLine(product, productName, description, quantity, unitPriceUntaxed, unitPriceTaxed, unit,
+					stockMove, trackingNumber);
+		} catch (Exception e) {
+			TraceBackService.trace(e);
+		}
+		return stockMoveLine;
 	}
```

Comment:
If we get an exception during stock move line creation, we should rollback and alert user. Else we get a partial stock move.